### PR TITLE
Add mypy-boto3-secretsmanager

### PR DESCRIPTION
**Pull request by qci-luke**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by qci-luke** [comment]
I am willing to be a maintainer.

**Comment by qci-luke** [comment]
Hmm... `build-locally` seems to work (or at least not throw errors) for me locally on windows. But it also doesn't seem to do much (vs. on osx and linux it has an additional build step).

File: recipes/mypy-boto3-secretsmanager/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.7
+    - pip
+  run:
+    - python >=3.7
+    - typing-extensions >=4.1.0
```
Comment (dopplershift) [inline recipes/mypy-boto3-secretsmanager/meta.yaml:23]: You need a dependency on `botocore`. I'm not sure why the upstream `setup.py` doesn't have it, but the package clearly imports it.

Comment (qci-luke) [reply recipes/mypy-boto3-secretsmanager/meta.yaml:23]: Aha, that's where to add it. Thank you!

**Review by dopplershift — APPROVED** [review]

**Pull request merged**